### PR TITLE
NY UP onboarding

### PR DESCRIPTION
**Pull request by victor-chaparro**
This PR adds updates to the NY receiver transforms and Simple Report's sender transforms. Some of the transforms added to Simple Report were being handled as receiver transforms for CA, but they should've been sender transforms in the first place. The integrations tests make sure that the updates made to the Simple Report sender transforms don't break CA. A follow up ticket #11289 will be needed to clean up the CA receiver transforms since a lot of them have been moved to sender transforms.

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. Check out and build the branch
2. run `./prime multiple-settings set -e local -i settings/staging/0155-NY-up-onboarding.yml`
3. Send a test report using the attached Postman collection
[NY.postman_collection.txt](https://github.com/CDCgov/prime-reportstream/files/12541742/NY.postman_collection.txt)
4. Verify that the request was accepted
5. Open the generated HL7 file in the local sftp folder.
6. Verify the file name contains NY receiver transforms specified in `NY-receiver-transforms.yml`

## Changes
- Updated Simple Report sender transforms
- Updated NY receiver transforms
- Added integration tests

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- #11231 

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #11289 

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: prime-router/metadata/fhir_transforms/senders/SimpleReport/simple-report-sender-transform.yml [hunk 1]
```diff
@@ -1,51 +1,222 @@
...
+    bundleProperty: '%patient.name.use'
+    value: [ '"official"' ]
+
+  # Covid pipeline populates this by looking at the hl70189 value-set version
+  - name: sr-patient-ethnic-group-version
+    bundleProperty: '%patient.extension("https://reportstream.cdc.gov/fhir/StructureDefinition/ethnic-group").value.coding.version'
+    value: [ '"2.9"' ]
+
+  # Covid pipeline populates this by looking at the hl70078 value-set version
+  - name: sr-observation-abnormal-flag
```
Comment (GilmoreA6) [inline prime-router/metadata/fhir_transforms/senders/SimpleReport/simple-report-sender-transform.yml:150]: Consider renaming sr-observation-abnormal-flag-version?

File: prime-router/metadata/fhir_transforms/senders/SimpleReport/simple-report-sender-transform.yml [hunk 2]
```diff
@@ -1,51 +1,222 @@
...
+    bundleProperty: '%patient.identifier.extension("https://reportstream.cdc.gov/fhir/StructureDefinition/assigning-facility-universal-id").value[x]'
+    value: [ '%diagnosticReport.basedOn.resolve().performer.resolve().identifier.value' ]
+
+  # Covid pipeline populates this by looking at the hl70005 value-set version
+  - name: sr-patient-race-version
+    bundleProperty: '%patient.extension("http://ibm.com/fhir/cdm/StructureDefinition/local-race-cd").value.coding.version'
+    value: [ '"2.5.1"' ]
+
+  # This is populated as a sender transform in the Covid pipeline
+  - name: sr-patient-name-user
```
Comment (GilmoreA6) [inline prime-router/metadata/fhir_transforms/senders/SimpleReport/simple-report-sender-transform.yml:140]: rename to sr-patient-name-use? or sr-patient-name-type?

File: prime-router/metadata/fhir_transforms/senders/SimpleReport/simple-report-sender-transform.yml [hunk 3]
```diff
@@ -1,51 +1,222 @@
...
-    value: [ '%resource.effective + 6 seconds' ]
-
-  - name: specimen-collection-time
+    bundleProperty: '%resource.type.coding.version'
+    value: [ '"2.67"' ]
+
+  # Covid pipeline populates this by looking at the sender-automation/aoe value-set version
+  # But some AOEs have different versions, we need the ability to lookup that version
+  # So the current solution is to move the sender-automation/aoe to the sender_automation_value_set_row lookup table
+  - name: test-condition
```
Comment (GilmoreA6) [inline prime-router/metadata/fhir_transforms/senders/SimpleReport/simple-report-sender-transform.yml:166]: potentially rename observation-code-version?

File: prime-router/src/testIntegration/resources/datatests/FHIR_to_HL7/sample_SR_1_20230302-0001.hl7 [hunk 4]
```diff
@@ -1,10 +1 @@
...
OBX|2|CE|80382-5^Flu A^LN||260373001^Detected^SCT|||A^Abnormal^HL70078^^^^2.7|||F||||11D1111111^CSV uploads^CLIA||^BD Veritor System for Rapid Detection of SARS-CoV-2 \T\ Flu A+B*^^^^^^^BD Veritor System for Rapid Detection of SARS-CoV-2 \T\ Flu A+B*||||||CSV uploads-11D1111111^^^^^CLIA&2.16.840.1.113883.4.7&ISO^XX^^^11D1111111|123 Main St^^^CA^94553^USA
OBX|3|CE|97099-6^COVID-19^LN||260415000^Not detected^SCT|||N^Normal^HL70078^^^^2.7|||F||||11D1111111^CSV uploads^CLIA||^BD Veritor System for Rapid Detection of SARS-CoV-2 \T\ Flu A+B*^^^^^^^BD Veritor System for Rapid Detection of SARS-CoV-2 \T\ Flu A+B*||||||CSV uploads-11D1111111^^^^^CLIA&2.16.840.1.113883.4.7&ISO^XX^^^11D1111111|123 Main St^^^CA^94553^USA
NTE|1|L|95419-8 Has symptoms related to condition of interest: No N HL70136
SPM|1|1519ae86-98f4-40d8-9780-d5211186e5b5&CSV uploads&11D1111111&CLIA^1519ae86-98f4-40d8-9780-d5211186e5b5&CSV uploads&11D1111111&CLIA||697989009^Anterior Nasal Swab^SCT^^^^2.67|||||||||||||20230330053236-0700
```
Comment (GilmoreA6) [inline prime-router/src/testIntegration/resources/datatests/FHIR_to_HL7/sample_SR_1_20230302-0001.hl7:1]: looks to be missing segment headers?

**Review by arnejduranovic — APPROVED** [review]
This looks good to me. Once the new mappings are completed, we may have to revisit these sender transforms and ensure all necessary extensions are set to ensure lossless FHIR -> HL7 conversion (Or SR can set them for us).

**Review by GilmoreA6 — APPROVED** [review]
LGTM. Made a few suggestions on element names to make them potentially more clear but that's a nitpicky item.

File: prime-router/src/testIntegration/resources/datatests/translation-test-config.csv [hunk 5]
```diff
@@ -22,8 +22,9 @@ HL7_to_FHIR_to_HL7/elims_29_5065302_35227_NoPII_CANCELED.hl7,,HL7_to_FHIR_to_HL7
...
-FHIR_to_HL7/sample_AOE_2_20230220-0001.fhir,,FHIR_to_HL7/sample_AOE_2_20230220-0001.hl7,metadata/hl7_mapping/receivers/STLTs/CA/CA-receiver-transform,HL7,PASS,"","","metadata/fhir_transforms/senders/SimpleReport/simple-report-sender-transform","",""
+FHIR_to_HL7/sample_AOE_2_20230220-0001.fhir,,FHIR_to_HL7/sample_AOE_2_20230220-0001.hl7,metadata/hl7_mapping/receivers/STLTs/CA/CA-receiver-transform,HL7,PASS,"","","metadata/fhir_transforms/senders/default-sender-transform","",""
 FHIR_to_HL7/sample_SR_1_20230302-0001.fhir,,FHIR_to_HL7/sample_SR_1_20230302-0001.hl7,metadata/hl7_mapping/receivers/STLTs/CA/CA-receiver-transform,HL7,PASS,"","","metadata/fhir_transforms/senders/SimpleReport/simple-report-sender-transform","",""
+FHIR_to_HL7/sample_SR_2_20231024-0002.fhir,,FHIR_to_HL7/sample_SR_2_20231024-0002.hl7,metadata/hl7_mapping/receivers/STLTs/NY/NY-receiver-transform,HL7,PASS,"","","metadata/fhir_transforms/senders/SimpleReport/simple-report-sender-transform","",""
```
Comment (jimmyfagan) [inline prime-router/src/testIntegration/resources/datatests/translation-test-config.csv:27]: Where is the SR_2 hl7? Wondering if it was missing from your commit

Comment (victor-chaparro) [reply prime-router/src/testIntegration/resources/datatests/translation-test-config.csv:27]: It is missing! Just added it, thanks!

**Review by oslynn — APPROVED** [review]

**Comment by oslynn** [comment]
It looks good, tested, and got the expected result.

**Pull request merged**